src/basicio.cpp

```diff
@@ -769,7 +776,14 @@ namespace Exiv2 {
                 typedef BOOL (WINAPI * ReplaceFileA_t)(LPCSTR, LPCSTR, LPCSTR, DWORD, LPVOID, LPVOID);
                 HMODULE hKernel = ::GetModuleHandleA("kernel32.dll");
                 if (hKernel) {
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-function-type"
```

Comment:
Are you sure that this is a good idea? Casting incompatible function types is usually a bug (and maybe undefined behavior).

Reply:
I had to remove that commit since the warning is anyways introduced in gcc-8 and was not present in previous versions of GCC. I read a bit about this issue in stackoverflow and I saw at least two PRs in other project just ignoring the warning. With MSVC and previous version of MinGW this seems to work correctly. Anyways, I have reconsidered this change and I think it is wiser to not touch this right now, with the changes going on for the RC1 release.